modules/appeals_api/spec/models/supplemental_claim_spec.rb

```diff
@@ -49,163 +50,248 @@
-        it { expect(supplemental_claim.veteran.state_code).to eq(address['stateCode']) }
-      end
+    it { expect(sc_with_nvc.outside_submission_window_error).to eq error }
+  end
 
-      describe 'mailing_address_country' do
-        it { expect(supplemental_claim.veteran.country_code).to eq(address['countryCodeISO2']) }
-      end
+  describe '#soc_opt_in' do
+    it { expect(sc_with_nvc.soc_opt_in).to be true }
+  end
 
-      describe 'zip_code_5' do
-        it { expect(supplemental_claim.veteran.zip_code_5).to eq(address['zipCode5']) }
-      end
+  describe '#form_5103_notice_acknowledged' do
+    it { expect(sc_with_nvc.form_5103_notice_acknowledged).to be true }
+  end
 
-      describe 'veteran_phone_data' do
-        it { expect(supplemental_claim.veteran_phone_data).to eq(veteran['phone']) }
-      end
+  describe '#date_signed' do
+    subject { sc_with_nvc.date_signed }
 
-      describe 'phone' do
-        it { expect(supplemental_claim.phone).to eq '+03-555-800-1111' }
-      end
+    it('matches json') do
+      expect(subject).to eq(
+        Time.now.in_time_zone(sc_with_nvc.signing_appellant.timezone).strftime('%m/%d/%Y')
+      )
     end
   end
 
-  describe 'V2' do
-    let(:supplemental_claim_v2) { create :extra_supplemental_claim }
-    let(:sc_veteran_only) { create(:minimal_supplemental_claim) }
+  describe '#stamp_text' do
+    let(:default_auth_headers) { fixture_as_json 'valid_200995_headers.json', version: 'v2' }
+    let(:form_data) { fixture_as_json 'valid_200995.json', version: 'v2' }
 
-    describe '#number_and_street' do
-      subject { supplemental_claim_v2.veteran.number_and_street }
+    it { expect(sc_with_nvc.stamp_text).to eq 'Doé - 6789' }
 
-      it { expect(subject).to eq('123 Main St Suite #1200 Box 4') }
-    end
+    it 'truncates the last name if too long' do
+      full_last_name = 'AAAAAAAAAAbbbbbbbbbbCCCCCCCCCCdddddddddd'
+      default_auth_headers['X-VA-Last-Name'] = full_last_name
 
-    describe '#city' do
-      subject { supplemental_claim_v2.veteran.city }
+      sc = AppealsApi::SupplementalClaim.new(auth_headers: default_auth_headers, form_data: form_data)
 
-      it { expect(subject).to eq('New York') }
+      expect(sc.stamp_text).to eq 'AAAAAAAAAAbbbbbbbbbbCCCCCCCCCCdd... - 6789'
     end
+  end
 
-    describe '#state_code' do
-      subject { supplemental_claim_v2.veteran.state_code }
+  describe '#update_status!' do
+    let(:supplemental_claim) { create(:supplemental_claim) }
 
-      it { expect(subject).to eq('NY') }
+    it 'error status' do
+      supplemental_claim.update_status!(status: 'error', code: 'code', detail: 'detail')
+
+      expect(supplemental_claim.status).to eq('error')
+      expect(supplemental_claim.code).to eq('code')
+      expect(supplemental_claim.detail).to eq('detail')
     end
 
-    describe '#country_code' do
-      subject { supplemental_claim_v2.veteran.country_code }
+    it 'other valid status' do
```

Comment:
"it updates the appeal with a valid status"

Reply:
Ugh sorry. I feel like this was a case of copy and past. Gonna clean 'er up.